int32[] data
---
int32[] ans